chore: Notify windows subprocess directly.

matta-aws opened the pull request:
### What was the problem/requirement? (What/Why)
The Notify signal for a windows process was sent from the `_signal_win_subprocess.py` script run as a subprocess.
We wish to remove the need to run it as another process.

### What was the solution? (How)
Call `_signal_win_subprocess.signal_process()` directly from OJD

### What is the impact of this change?
One less subprocess run by OJD

### How was this change tested?
Unit tests

### Was this change documented?
No
### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

ddneilson commented on src/openjd/sessions/_subprocess.py:
```diff
@@ -326,20 +323,12 @@ def _windows_sigbreak_subprocess(self) -> None:
-                    result.stdout.decode("utf-8"),
-                )
+            # otherwise we need to attach to the _process console in order to signal it
+            try:
```
Can you try an experiment? Run this within the `openjd-cli`, but first modify the CLI to trap CTRL_BREAK_EVENT. I tried something similar to this when experimenting with https://github.com/OpenJobDescription/openjd-sessions-for-python/pull/97 and was finding that my current process was getting the signal despite detaching from the console (I had also modified this code to send a CTRL_C in addition).

matta-aws replied:
Ctrl-c, btw, cannot be limited to a process group, while ctrl-break can.
https://learn.microsoft.com/en-us/windows/console/generateconsolectrlevent?redirectedfrom=MSDN

jericht commented:
This will need to be updated to remove usage of pywin32 after we merge https://github.com/OpenJobDescription/openjd-sessions-for-python/pull/97

matta-aws committed:
chore: Notify windows subprocess directly

Problem
The Notify signal for a windows process was sent from the _signal_win_subprocess.py script run as a subprocess. We wish to remove the need to run it as another process.

Solution
Call signal_process directly

Signed-off-by: Authement <maauth@amazon.com>

matta-aws commented:
Rebased on mainline.  As signal_process now uses ctypes win apis, call that rather than use the pywin32 apis.

jericht commented:
Spoke with @ddneilson and we've decided to just close this PR since the existing implementation is fine.
